### PR TITLE
Initialize Prices and Price Types closes #1925

### LINKED ISSUE #1925: Fetch price types to be used for ticket mutation.
Price types are needed for price modifiers for tickets. We need to fetch them to be able to create ticket price modifiers.

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Completes #1925 and provides foundation for #1929 

## Checklist

* [ ] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.
* [ ] My code accounts for when the site is in Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

File: assets/prototype/domain/eventEditor/EventEditor.js [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
 
 const EventEditor = ({ eventId }) => {
 	const { datetimes, datetimeError, loadingDates, tickets, ticketError, loadingTickets } = useInitQueries({
 		eventId,
 	});
+
+	useFetchPriceTypes();
+
+	const ticketIn = tickets.map(({ id }) => id);
+	useFetchPrices({ ticketIn });
```
Comment (tn3rb) [inline assets/prototype/domain/eventEditor/EventEditor.js]: why not add this to `useInitQueries` as well?
we have the `tickets` array there and this component is primarily concerned with display

File: assets/prototype/domain/eventEditor/containers/mutations/prices.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+export const CREATE_PRICE = gql`
+	mutation createPrice($input: CreatePriceInput!) {
+		createPrice(input: $input) {
+			price {
+				...priceAttributes
+			}
+		}
+	}
+	${PRICE_ATTRIBUTES}
```
Comment (tn3rb) [inline assets/prototype/domain/eventEditor/containers/mutations/prices.js]: nice 👍

File: assets/prototype/domain/eventEditor/containers/queries/priceTypes.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+export const PRICE_TYPE_ATTRIBUTES = gql`
+	fragment priceTypeAttributes on PriceType {
+		id
+		priceTypeId
+		name
+		baseType
+		isPercent
+		isDiscount
+	}
+`;
```
Comment (tn3rb) [inline assets/prototype/domain/eventEditor/containers/queries/priceTypes.js]: oh yeah! really nice! 👍 
we need to do this for other queries too

File: assets/prototype/domain/eventEditor/containers/queries/useFetchPrices.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	const toaster = useToaster();
+	const toasterMessage = 'initializing prices';
+
+	const { data, error, loading } = useQuery(GET_PRICES, {
+		variables: {
+			where: {
+				ticketIn,
+			},
+		},
+		skip: !ticketIn.length, // do not fetch if we don't have any tickets
```
Comment (tn3rb) [inline assets/prototype/domain/eventEditor/containers/queries/useFetchPrices.js]: cool, glad we pivoted to using Apollo!

File: assets/prototype/infrastructure/services/apollo/Apollo.js [hunk 5]
```diff
@@ -7,11 +7,17 @@ import get from 'lodash/get';
 const { graphqlEndpoint } = window;
 const nonce = get(window, ['eejsdata', 'data', 'eejs_api_nonce']);
 
+const getResolver = (type) => {
+	return (_, args, { getCacheKey }) => getCacheKey({ __typename: type, id: args.id });
+};
```
Comment (tn3rb) [inline assets/prototype/infrastructure/services/apollo/Apollo.js]: nice 👍

File: assets/prototype/domain/eventEditor/EventEditor.js (hunk 1 above)
Comment (manzoorwanijk) [reply assets/prototype/domain/eventEditor/EventEditor.js]: I initially did that but it was getting bigger and I thought to keep them separate. I think we should create separate hooks - `useFetchDatetimes`, `useFetchTickets`, `useFetchPriceTypes`, `useFetchPrices` etc. and then combine them inside `useInitQueries`. Every hook will handle its dependencies itself.

**Pull request merged**